argoverse/utils/sim2.py

```diff
@@ -0,0 +1,95 @@
+"""
+Utility for 2d rigid body transformations with scaling.
+
+Refs:
+    http://ethaneade.com/lie_groups.pdf
+    https://github.com/borglab/gtsam/blob/develop/gtsam_unstable/geometry/Similarity3.h
+"""
+
+from typing import Union
+
+import numpy as np
+
+from argoverse.utils.helpers import assert_np_array_shape
+
+
+class Sim2:
```

Comment:
Should we potentially create a `LieGroup` super-class? Seems like there would be a lot of overlapping functionality between these different transformations.

Reply:
Additionally, I think sub-classing `numpy.ndarray` might be something we should evaluate: https://numpy.org/doc/stable/user/basics.subclassing.html. We could implement matrix multiplication directly on the object itself (in addition to transformation methods).

Reply:
Haven't seen the subclassing before on `numpy.ndarray` -- would it make things more efficient? i'm not sure it would be more readable since there are a few different types of matrices here (3x3, or 3x1, or 4x4)

Reply:
I don't think we'd see any performance improvement. It would mainly be syntactic sugar, but I guess it could get confusing and lead to unintentional behavior. Idk, I just think that transformation syntax gets pretty verbose.

Reply:
Another thought -- LieGroups seems like they should almost be an entirely separate library.